docs/enhancements/install-sno-using-ibi.md

```diff
@@ -0,0 +1,133 @@
+
+Currently, installing an SNO cluster using the Assisted-Installer takes up to
+~40min using the current flow which installs OCP from the ground-up.
+
+We can reduce the installation time to ~10min by leveraging the Image-Based
+installation method, and as a way to increase the success rate of SNO
+installations as we restore previously working setup.
+
+This enhancement describes the changes to bring to the Assisted-Installer in
+order to perform SNO installation using the Image-Based installation method.
+
+## Motivation
+
+The Image-Based installation brings the following advantages:
+- Users will experience faster SNO installations
+- Increased success rate installations since we restore the state of a
+  previously working SNO setup
+
+### Goals
+
+- Provide an option in Assisted-Installer to allow a user to install a SNO with
+  the Image-Based Installation method
+
+### Non-Goals
+
+- Focus is on Assisted-Installer in SaaS environment, we don’t aim to support
+  other environments at this point
+
+## Proposal
+
+### As a user, I want to install a SNO using the IBI method
+
+#### REST API changes
+
+A new flag will be added to the cluster object [TBD], along with
+extra-parameters specific to IBI [TBD].
+
+`supported-versions` API with be amended to return only the version of OCP
+available for installation with IBI.
+
+#### Feature support
+
+A new feature support will be created for the IBI installation. 
+The feature will be gated by an OCM capability until we make it available to
+all users.
+
+#### Installation
+
+When the user will hit the installation button, the `assisted-installer` will
+restore the image seed on the disk selected for installation:
+- Install RHCOS using `coreos-installer`
+- Partition the disk to create a dedicated `/var/lib/containers`
+- Grow `/` partition
+- Mount `/`, `/boot` and `/var/lib/container`
+- Download and restore the seed on the disk using `lca-cli`
+
+[Reference](https://github.com/openshift-kni/lifecycle-agent/blob/main/ib-cli/installationiso/data/install-rhcos-and-restore-seed.sh)
+
+#### Configuration
+
+After the image seed is restored on the disk, the `assisted-installer` will
+drop the configuration in `/opt/openshift`:
+- A
+  [manifest.json](https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/post-pivot-configuration.md#user-specifications)
+  with: 
+   - Hostname: will be picked from the inventory of host
+   - Cluster name: will be picked from the cluster definition 
+   - BaseDomain: will be picked from the cluster definition 
+   - ClusterID: will be picked from the cluster definition 
+   - SSHkey: will be provided by the user
+   - KubeadminPasswordHash: will be generated by Assisted-Installer and
+     provided to the user
+- Pull secret
+- assisted-controller deployment manifest, it will tell us if the installation
+  succeeded, and provide a way to get installation logs?
+- generate cryptographic materials in order to provide a kubeconfig file to the
+  user
```

Comment:
@omertuc brought up a point recently about us baking the private keys into the ISO, it might be wise to have these go over an API request from the agent rather than sit in an artifact that the user might misplace.

Reply:
Yeah, but let's keep things simple for now and have that in the back of ours minds as a ticket, until we figure out the security side of things and whether this is a threat that's relevant to protect from